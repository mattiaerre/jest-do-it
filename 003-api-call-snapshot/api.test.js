const api = require('./api');

const scenarios = [
  {
    name: 'Luke Skywalker',
    id: 1
  },
  {
    name: 'Darth Vader',
    id: 4
  }
];

scenarios.forEach((scenario) => {
  test(scenario.name, async () => {
    const data = await api(scenario.id);
    expect(data).toMatchSnapshot();
  });
});
