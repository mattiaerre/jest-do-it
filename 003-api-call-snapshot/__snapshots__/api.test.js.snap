// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Darth Vader 1`] = `
Object {
  "birth_year": "41.9BBY",
  "created": "2014-12-10T15:18:20.704000Z",
  "edited": "2014-12-20T21:17:50.313000Z",
  "eye_color": "yellow",
  "films": Array [
    "http://swapi.co/api/films/2/",
    "http://swapi.co/api/films/6/",
    "http://swapi.co/api/films/3/",
    "http://swapi.co/api/films/1/",
  ],
  "gender": "male",
  "hair_color": "none",
  "height": "202",
  "homeworld": "http://swapi.co/api/planets/1/",
  "mass": "136",
  "name": "Darth Vader",
  "skin_color": "white",
  "species": Array [
    "http://swapi.co/api/species/1/",
  ],
  "starships": Array [
    "http://swapi.co/api/starships/13/",
  ],
  "url": "http://swapi.co/api/people/4/",
  "vehicles": Array [],
}
`;

exports[`Luke Skywalker 1`] = `
Object {
  "birth_year": "19BBY",
  "created": "2014-12-09T13:50:51.644000Z",
  "edited": "2014-12-20T21:17:56.891000Z",
  "eye_color": "blue",
  "films": Array [
    "http://swapi.co/api/films/2/",
    "http://swapi.co/api/films/6/",
    "http://swapi.co/api/films/3/",
    "http://swapi.co/api/films/1/",
    "http://swapi.co/api/films/7/",
  ],
  "gender": "male",
  "hair_color": "blond",
  "height": "172",
  "homeworld": "http://swapi.co/api/planets/1/",
  "mass": "77",
  "name": "Luke Skywalker",
  "skin_color": "fair",
  "species": Array [
    "http://swapi.co/api/species/1/",
  ],
  "starships": Array [
    "http://swapi.co/api/starships/12/",
    "http://swapi.co/api/starships/22/",
  ],
  "url": "http://swapi.co/api/people/1/",
  "vehicles": Array [
    "http://swapi.co/api/vehicles/14/",
    "http://swapi.co/api/vehicles/30/",
  ],
}
`;
